src/item.h

```diff
@@ -221,17 +238,49 @@ class item : public JsonSerializer, public JsonDeserializer
          * items that stack together (@ref stacks_with).
          */
         bool merge_charges( const item &rhs );
- void put_in(item payload);
- void add_rain_to_container(bool acid, int charges = 1);
 
  int weight() const;
 
  int precise_unit_volume() const;
  int volume(bool unit_value=false, bool precise_value=false) const;
- int volume_contained() const;
- int attack_time() const;
- int damage_bash() const;
- int damage_cut() const;
+
+    /**
+     * @name Melee
+     *
+     * The functions here assume the item is used in melee, even if's a gun or not a weapon at
+     * all. Because the functions apply to all types of items, several of the is_* functions here
+     * may return true for the same item. This only indicates that it can be used in various ways.
+     */
+    /*@{*/
+    /**
+     * Number of moves (@ref Creature::moves) that a single attack take.
+     */
+    int attack_time() const;
+    /**
+     * Damage of type @ref DT_BASH that is caused by using this item as melee weapon.
+     */
+    int damage_bash() const;
+    /**
+     * Damage of type @ref DT_CUT that is caused by using this item as melee weapon.
+     */
+    int damage_cut() const;
+    /**
+     * Whether the character needs both hands to wield this item.
+     */
+    // TODO: make a reference. Make a const reference.
+    bool is_two_handed(player *u);
+    /** The weapon is considered a suitable melee weapon. */
+    bool is_weap() const;
+    /** The weapon inflicts at least a certain minimal basing damage. */
```

Comment:
> Why not just explicitly say how much? This number probably won't change soon.

The number should be of no interest to the read. In fact, it's an implementation detail, the reader of the documentation should not depend on that behavior.

But the wording is bit unclear (and in its current form not much better than showing the number). I changed it to show what the functions intention is.